### PR TITLE
Move responsibilty for ensuring greeting parameters are set to `ensureGreetingParamsAreSet`

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
This just extends https://github.com/civicrm/civicrm-core/pull/21911 so we can rely on `ensureGreetingParamsAreSet` to ensure greeting params are set - and not mop them up post save

Before
----------------------------------------
_What is the old user-interface or technical-contract (as appropriate)?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

After
----------------------------------------
_What changed? What is new old user-interface or technical-contract?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

Technical Details
----------------------------------------
_If the PR involves technical details/changes/considerations which would not be manifest to a casual developer skimming the above sections, please describe the details here._

Comments
----------------------------------------
_Anything else you would like the reviewer to note_

### CODE REVIEW

File: CRM/Contact/BAO/Contact.php [hunk 1]
```diff
@@ -541,6 +541,10 @@ public static function ensureGreetingParamsAreSet(&$params) {
         //  If we are setting it to null then null out the display field.
         $params[$greetingIndex . '_display'] = 'null';
       }
+      if ((int) $params[$greetingParam] !== CRM_Core_PseudoConstant::getKey('CRM_Contact_BAO_Contact', $greetingParam, 'Customized')) {
```
Comment (colemanw) [inline CRM/Contact/BAO/Contact.php:544]: I'm curious about this strict operator. I'm not sure the `getKey` function can be relied upon to return strict integer types; it's an old function that afaik follow's PHP's default sloppy equivocation of numbers and strings.

Comment (eileenmcnaughton) [reply CRM/Contact/BAO/Contact.php]: @colemanw I hate non-strict but I pulled it back in case it's unreliable

**Comment by eileenmcnaughton** [comment]
@colemanw any chance you could check this one again - the test fails are unrelated

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by seamuslee001** [comment]
This seems fine to me merging

**Pull request merged**